packages/expo/cli/start/server/middleware/ExpoGoManifestHandlerMiddleware.ts

```diff
@@ -0,0 +1,134 @@
+import { ManifestMiddleware, ParsedHeaders } from './ManifestMiddleware';
+import {
+  parsePlatformHeader,
+  assertMissingRuntimePlatform,
+  assertRuntimePlatform,
+} from './resolvePlatform';
+import { ServerHeaders, ServerRequest } from './server.types';
+
+export class ExpoGoManifestHandlerMiddleware extends ManifestMiddleware {
+  public getParsedHeaders(req: ServerRequest): ParsedHeaders {
+    const platform = parsePlatformHeader(req);
+    assertMissingRuntimePlatform(platform);
+    assertRuntimePlatform(platform);
+
+    return {
+      platform,
+      acceptSignature: !!req.headers['expo-accept-signature'],
+      hostname: stripPort(req.headers['host']),
+    };
+  }
+
+  protected getDefaultResponseHeaders(): Map<string, any> {
+    const headers = new Map<string, any>();
+    // set required headers for Expo Updates manifest specification
+    headers.set('expo-protocol-version', 0);
+    headers.set('expo-sfv-version', 0);
+    headers.set('cache-control', 'private, max-age=0');
+    headers.set('content-type', 'application/json');
+    return headers;
+  }
+
+  public async _getManifestResponseAsync(requestOptions: ParsedHeaders): Promise<{
+    body: string;
+    version: string;
+    headers: ServerHeaders;
+  }> {
+    const { exp, hostUri, expoGoConfig, bundleUrl } = await this._resolveProjectSettingsAsync(
+      requestOptions
+    );
+
+    const runtimeVersion = Updates.getRuntimeVersion(
+      { ...exp, runtimeVersion: exp.runtimeVersion ?? { policy: 'sdkVersion' } },
+      requestOptions.platform
+    );
+
+    const easProjectId = exp.extra?.eas?.projectId;
+    const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync(easProjectId);
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    if (!shouldUseAnonymousManifest) {
+      assert(easProjectId);
+    }
+    const scopeKey = shouldUseAnonymousManifest
+      ? `@${ANONYMOUS_USERNAME}/${exp.slug}-${userAnonymousIdentifier}`
+      : await this.getScopeKeyForProjectIdAsync(easProjectId);
+
+    const expoUpdatesManifest = {
+      id: uuidv4(),
+      createdAt: new Date().toISOString(),
+      runtimeVersion,
+      launchAsset: {
+        key: 'bundle',
+        contentType: 'application/javascript',
+        url: bundleUrl,
+      },
+      assets: [], // assets are not used in development
+      metadata: {}, // required for the client to detect that this is an expo-updates manifest
+      extra: {
+        eas: {
+          projectId: easProjectId ?? undefined,
+        },
+        expoClient: {
+          ...exp,
+          hostUri,
+        },
+        expoGo: expoGoConfig,
+        scopeKey,
+      },
+    };
+
+    const headers = this.getDefaultResponseHeaders();
+    if (requestOptions.acceptSignature && !shouldUseAnonymousManifest) {
+      // TODO: Should this also be memoized?
+      const manifestSignature = await signExpoGoManifestAsync(expoUpdatesManifest);
```

Comment:
Memoizing it for the duration of the CLI process should be fine.

This said, we're working towards making it so signing requires an internet connection only every now and then in order to get a signing certificate that's valid for, say, a week. Once you have a valid signing certificate, you'll be able to sign development manifests while offline and be able to access scoped data in Expo Go without `@anonymous` or unsigned manifests.